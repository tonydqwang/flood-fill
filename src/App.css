.MuiCard-root {
  margin: 20px 0 20px 20px;
  padding: 20px;
}

.main {
  display: flex;
  justify-content: center;
}