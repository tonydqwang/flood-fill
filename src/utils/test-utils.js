import { render } from '@testing-library/react';
import { AppCtxProvider } from '../components/app-context';
import { act } from 'react-dom/test-utils';

const AllTheProviders = ({ chilren }) => {
  return (
    <AppCtxProvider>
      {chilren}
    </AppCtxProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

export { 
  customRender as render,
}