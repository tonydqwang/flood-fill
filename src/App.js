import React, { useContext } from 'react';
import 'devextreme/dist/css/dx.common.css';
import 'devextreme/dist/css/dx.light.css';
import './App.css';
import Header from './components/header';
import Canvas from './components/canvas';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';

function App() {
  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
  const theme = React.useMemo(
    () =>
      createMuiTheme({
        palette: {
          type: prefersDarkMode ? 'dark' : 'light',
        },
      }),
    [prefersDarkMode],
  );

  const genData = (sizeX, sizeY) => {
    const data = Array(sizeX, sizeY)
    for (var y = 0; y < sizeY; y++) {
      for (var x = 0; x < sizeX; x++) {
        data[y * sizeX + x] = { x, y, color: '#ffffff' };
      }
    }
    return data;
  }

  const canvasSizeX = 5;
  const canvasSizeY = 5;
  const activeColor = '#ffffff';
  const canvasData = genData(canvasSizeX, canvasSizeY);

  return (
    <>
      <ThemeProvider theme={theme}>
        <CssBaseline/>
        <Header />
        <main>
          <div className='main'>
            <Canvas 
              data={canvasData} 
              sizeX={canvasSizeX} 
              sizeY={canvasSizeY}
              color={activeColor}
            />
          </div>
        </main>
      </ThemeProvider>
    </>
  );
}

export default App;