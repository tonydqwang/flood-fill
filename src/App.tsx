import React, { useContext } from 'react';
import 'devextreme/dist/css/dx.common.css';
import 'devextreme/dist/css/dx.light.css';
import './App.css';
import { AppContext } from './components/app-context';
import Header from './components/header';
import Canvas from './components/canvas';
import Control from './components/control';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';

function App() {
  const appContext = useContext(AppContext);
  const { state, handlers } = appContext;
  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
  const theme = React.useMemo(
    () =>
      createMuiTheme({
        palette: {
          type: prefersDarkMode ? 'dark' : 'light',
        },
      }),
    [prefersDarkMode],
  );

  return (
    <>
      <ThemeProvider theme={theme}>
        <CssBaseline/>
        <Header />
        <main>
          <div className='main'>
            <Control 
              sizeX={state.canvasSizeX} 
              sizeY={state.canvasSizeY}
              handleSizeXChange={handlers.handleSizeXChange}
              handleSizeYChange={handlers.handleSizeYChange}
              tool={state.activeTool}
              handleToolChange={handlers.handleToolChange}
              color={state.activeColor}
              handleColorChange={handlers.handleColorChange}
            />
            <Canvas 
              data={state.canvasData} 
              sizeX={state.canvasSizeX} 
              sizeY={state.canvasSizeY}
              handleClick={handlers.handleClick}
            />
          </div>
        </main>
      </ThemeProvider>
    </>
  );
}

export default App;