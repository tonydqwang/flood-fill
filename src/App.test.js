import { render } from './utils/test-utils';
import App from './App';

test('renders default app', async () => {
  const { baseElement } = await render(<App />);
  expect(baseElement).toMatchSnapshot();
});
