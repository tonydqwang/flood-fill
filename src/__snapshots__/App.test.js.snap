// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<body>
  <div>
    <header>
      <div
        class="header"
      >
        <p>
          / Welcome to Flood Fill
        </p>
      </div>
    </header>
    <main>
      <div
        class="main"
      >
        <div
          class="MuiPaper-root MuiCard-root canvas MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="canvas--row"
          >
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
          </div>
          <div
            class="canvas--row"
          >
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
          </div>
          <div
            class="canvas--row"
          >
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
          </div>
          <div
            class="canvas--row"
          >
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
          </div>
          <div
            class="canvas--row"
          >
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
            <div
              class="canvas--cell"
              style="background-color: rgb(255, 255, 255);"
            />
          </div>
        </div>
      </div>
    </main>
  </div>
</body>
`;
